reaktive/src/commonTest/kotlin/com/badoo/reaktive/completable/DoOnBeforeCompleteTest.kt

```diff
@@ -0,0 +1,33 @@
+package com.badoo.reaktive.completable
+
+import com.badoo.reaktive.test.completable.DefaultCompletableObserver
+import com.badoo.reaktive.test.completable.TestCompletable
+import com.badoo.reaktive.test.utils.SafeMutableList
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class DoOnBeforeCompleteTest
+    : CompletableToCompletableTests by CompletableToCompletableTests({ doOnBeforeComplete {} }) {
+
+    @Test
+    fun calls_action_before_completion() {
+        val upstream = TestCompletable()
+        val callOrder = SafeMutableList<String>()
+
+        upstream
+            .doOnBeforeComplete {
+                callOrder += "action"
+            }
+            .subscribe(
+                object : DefaultCompletableObserver {
+                    override fun onComplete() {
+                        callOrder += "onComplete"
+                    }
+                }
+            )
+
+        upstream.onComplete()
+
+        assertEquals(listOf("action", "onComplete"), callOrder.items)
+    }
+}
```

Comment:
Can you add at least one test case, when "doOnBeforeXXX" should not be called? For example: do not call `onComplete` in this test and verify, that `doOnBeforeComplete` block was not called. `doOnBeforeComplete` can have broken behaviour and be executed when subscription was created. In this case test still passes.